acceptance/tests/wan-federation/wan_federation_test.go

```diff
@@ -190,11 +209,259 @@ func TestWANFederation(t *testing.T) {
+
+				"server.extraVolumes[0].type":          "secret",
+				"server.extraVolumes[0].name":          federationSecretName,
+				"server.extraVolumes[0].load":          "true",
+				"server.extraVolumes[0].items[0].key":  "serverConfigJSON",
+				"server.extraVolumes[0].items[0].path": "config.json",
+
+				"connectInject.enabled":  "true",
+				"connectInject.replicas": "1",
+
+				"meshGateway.enabled":  "true",
+				"meshGateway.replicas": "1",
+
+				"global.enableConsulNamespaces":               "true",
+				"connectInject.consulNamespaces.mirroringK8S": "true",
+			}
+
+			if c.secure {
+				secondaryHelmValues["global.acls.replicationToken.secretName"] = federationSecretName
+				secondaryHelmValues["global.acls.replicationToken.secretKey"] = "replicationToken"
+				secondaryHelmValues["global.federation.k8sAuthMethodHost"] = k8sAuthMethodHost
+				secondaryHelmValues["global.federation.primaryDatacenter"] = primaryDatacenter
+			}
+
+			if cfg.UseKind {
+				secondaryHelmValues["meshGateway.service.type"] = "NodePort"
+				secondaryHelmValues["meshGateway.service.nodePort"] = "30000"
+			}
+
+			// Install the secondary consul cluster in the secondary kubernetes context
+			secondaryConsulCluster := consul.NewHelmCluster(t, secondaryHelmValues, secondaryContext, cfg, releaseName)
+			secondaryConsulCluster.Create(t)
+
+			primaryClient, _ := primaryConsulCluster.SetupConsulClient(t, c.secure)
+			secondaryClient, _ := secondaryConsulCluster.SetupConsulClient(t, c.secure)
+
+			// Verify federation between servers
+			logger.Log(t, "Verifying federation was successful")
+			helpers.VerifyFederation(t, primaryClient, secondaryClient, releaseName, c.secure)
+
+			// Create a ProxyDefaults resource to configure services to use the mesh
+			// gateways.
+			logger.Log(t, "Creating proxy-defaults config")
+			kustomizeDir := "../fixtures/bases/mesh-gateway"
+			k8s.KubectlApplyK(t, secondaryContext.KubectlOptions(t), kustomizeDir)
+			helpers.Cleanup(t, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
+				k8s.KubectlDeleteK(t, secondaryContext.KubectlOptions(t), kustomizeDir)
+			})
+
+			primaryHelper := connhelper.ConnectHelper{
+				Secure:          c.secure,
+				ReleaseName:     releaseName,
+				Ctx:             primaryContext,
+				UseAppNamespace: false,
+				Cfg:             cfg,
+				ConsulClient:    primaryClient,
+			}
+			secondaryHelper := connhelper.ConnectHelper{
+				Secure:          c.secure,
+				ReleaseName:     releaseName,
+				Ctx:             secondaryContext,
+				UseAppNamespace: false,
+				Cfg:             cfg,
+				ConsulClient:    secondaryClient,
+			}
+
+			// Create Namespaces
+			primaryNamespaceOpts := primaryHelper.Ctx.KubectlOptionsForNamespace(primaryNamespace)
+			primaryHelper.SetupNamespace(t, primaryNamespaceOpts.Namespace)
+			primarySecondaryNamepsaceOpts := primaryHelper.Ctx.KubectlOptionsForNamespace(secondaryNamespace)
+			primaryHelper.SetupNamespace(t, primarySecondaryNamepsaceOpts.Namespace)
```

Comment:
~ It might help to briefly comment here similar to below that this is a second namespace in the primary DC... I think the "primary" and "secondary" terms here might be a bit hard to follow to someone reading the test for the first time